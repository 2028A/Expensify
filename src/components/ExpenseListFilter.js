import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter,setSortByDate,setSortByAmount,setStartDate,setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';


export class ExpenseListFilters extends React.Component{
  constructor(props){
    super(props);
    this.state={
      calendarFocused:null
    }
  }

 
onDatesChange=({startDate,endDate})=>{
  this.props.setStartDate(startDate);
  this.props.setEndDate(endDate);
};
onFocusChange=(calendarFocused)=>{
    this.setState(()=>({calendarFocused}));
};

onSortChange=(e)=>{
  if(e.target.value==="date")
  {
    this.props.setSortByDate();
  }
  else
  {
    this.props.setSortByAmount();
  }
}
 
onTextChange=(e)=>{
   this.props.setTextFilter(e.target.value);
};
 render(){
   return(
    <div className="content-container">
      <div className="input-group">
      <div className="input-group__item">
        <input type="text" 
        className="text-input"
        placeholder="Search expenses"
        value={this.props.filters.text}
            onChange={this.onTextChange}
        />
        </div>
        <div className="input-group__item">
        <select className="select"
        onChange={this.onSortChange}
            value={this.props.filters.sortBy}>
            <option value="date">Date</option>
            <option value='amount'>Amount</option>
        </select>
        </div>
        <div className="input-group__item">
    <DateRangePicker 
      startDate={this.props.filters.startDate}
      endDate={this.props.filters.endDate}
      onDatesChange={this.onDatesChange}
      focusedInput={this.state.calendarFocused}
      onFocusChange={this.onFocusChange}
      showClearDates={true}
      numberOfMonths={1}
      isOutsideRange={()=>{false}}
    />
    </div>
      </div>
    </div>
    
)};

}
const mapStoreToProps=(state)=>({
         filters:state.filters
    }
);
const mapDispatchToProps=(dispatch)=>({
  setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
  setEndDate:(endDate)=>dispatch(setEndDate(endDate)),
  setSortByDate:()=>dispatch(setSortByDate()),
  setSortByAmount:()=>dispatch(setSortByAmount()),
  setTextFilter:(text)=>dispatch(setTextFilter(text))
});
export default connect(
    mapStoreToProps,mapDispatchToProps)(ExpenseListFilters);

