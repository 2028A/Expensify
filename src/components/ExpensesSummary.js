import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary=({expensesCount,expensesTotal})=>{
    const expenseWord=expensesCount<=1 ? 'expense':'expenses';
    const formmatedExpenseTotal=numeral(expensesTotal/100).format('$0,0.00');
    return (
    <div className="page-header">
    <div className="content-container">
    <h1 className="page-header__title">View <span>{expensesCount} {expenseWord}</span> totaling <span>{formmatedExpenseTotal}</span></h1>
    <div className="page-header__actions">
        <Link to="/create" className="button">Add Expense</Link>
    </div>
    </div>
        
    </div>
    );
}

const mapStateToProps=(state)=>{
    const visibleExpenses=selectExpenses(state.expenses,state.filters);
    return {
        expensesCount:visibleExpenses.length,
        expensesTotal:selectExpensesTotal(visibleExpenses)
    }
    
};

export default connect(mapStateToProps)(ExpensesSummary);