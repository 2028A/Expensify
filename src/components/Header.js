import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {startLogout} from '../actions/auth';

export const Header=({startLogout})=>(
    <header className="header">
       <div className="content-container">
       <div className="header__content">
        <Link to="/dashboard" className="header__title">
        <h1 >Expensify</h1></Link>
        
        <button onClick={startLogout}>Logout</button>
        </div>
        </div>
    </header>
    );


    const mapDispatchToProps=(dispath)=>({
        startLogout:()=>dispath(startLogout())
      });
      
      export default connect(undefined,mapDispatchToProps)(Header);
      