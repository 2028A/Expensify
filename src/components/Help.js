import React from 'react';

const ExpensifyHelpPage=()=>(
    <div>This is Enpensify help Page</div>
    );

export default ExpensifyHelpPage;
