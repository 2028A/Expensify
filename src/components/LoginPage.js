import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage=({startLogin})=>(
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Expensify App</h1>
        <p>Is time to get your expenses under control</p>
      
        <button className="button" onClick={startLogin}>LogIn</button>
        </div>
    </div>
);

const mapDispatchToProps=(dispath)=>({
  startLogin:()=>dispath(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);
