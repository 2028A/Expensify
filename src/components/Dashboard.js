import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import ExpensesSummary from './ExpensesSummary';
const ExpensifyDashboardPage=()=>(
    <div>
        <ExpenseListFilter />
        <ExpenseList />
        <ExpensesSummary />
    </div>
);

export default ExpensifyDashboardPage;