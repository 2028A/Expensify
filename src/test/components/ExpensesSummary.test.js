import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
//react-test-renderer

test('Should render expenses Summary with expenses',()=>{
    const wrapper=shallow(<ExpensesSummary expensesCount={4} expensesTotal={5454544} />);
    expect(wrapper).toMatchSnapshot();
   
})

test('Should render expenses Summary with one expenses',()=>{
    const wrapper=shallow(<ExpensesSummary expensesCount={1} expensesTotal={54544}  />);
    expect(wrapper).toMatchSnapshot();
   
})