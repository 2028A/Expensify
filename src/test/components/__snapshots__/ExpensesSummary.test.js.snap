// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenses Summary with expenses 1`] = `
<div>
  <h1>
    View 
    4
     
    expenses
     totaling 
    $54,545.44
  </h1>
</div>
`;

exports[`Should render expenses Summary with one expenses 1`] = `
<div>
  <h1>
    View 
    1
     
    expense
     totaling 
    $545.44
  </h1>
</div>
`;
